stake-o-matic/src/confirmed_block_cache.rs

```diff
@@ -0,0 +1,203 @@
+        let extension = parts.next();
+        extension
+            .zip(slot_range)
+            .and_then(|(extension, slot_range)| {
+                if extension == "json" {
+                    Some(slot_range)
+                } else {
+                    None
+                }
+            })
+            .and_then(|slot_range| {
+                let mut parts = slot_range.splitn(2, '-');
+                let start = parts.next().and_then(|p| p.parse::<Slot>().ok());
+                let end = parts.next().and_then(|p| p.parse::<Slot>().ok());
+                start.zip(end).map(|(start, end)| start..end)
+            })
+    }
+
+    pub fn from_pathbuf(path: PathBuf) -> Option<Self> {
+        path.file_name()
+            .and_then(|n| n.to_str())
+            .and_then(|n| Self::parse_filename(n))
+            .map(|slots| Self { slots, path })
+    }
+
+    pub fn path(&self) -> &Path {
+        &self.path
+    }
+}
+
+const CHUNK_SIZE: u64 = 2500;
+const CONFIG_FILENAME: &str = "config.yaml";
+
+#[derive(Debug, Deserialize, Serialize)]
+struct Config {
+    chunk_size: u64,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Self {
+            chunk_size: CHUNK_SIZE,
+        }
+    }
+}
+
+pub struct ConfirmedBlockCache {
+    rpc_client: RpcClient,
+    base_path: PathBuf,
+    entries: RefCell<Vec<Entry>>,
+    config: Config,
+}
+
+impl ConfirmedBlockCache {
+    fn store_config<P: AsRef<Path>>(config_path: P, config: &Config) -> io::Result<()> {
+        let config_path = config_path.as_ref();
+        let file = File::create(config_path)?;
+        serde_yaml::to_writer(file, config).map_err(|e| {
+            io::Error::new(
+                io::ErrorKind::Other,
+                format!(
+                    "error: cannot store config `{}`: {:?}",
+                    config_path.to_string_lossy(),
+                    e,
+                ),
+            )
+        })
+    }
+
+    fn load_config<P: AsRef<Path>>(config_path: P) -> io::Result<Config> {
+        let config_path = config_path.as_ref();
+        let file = File::open(config_path)?;
+        serde_yaml::from_reader(file).map_err(|e| {
+            io::Error::new(
+                io::ErrorKind::Other,
+                format!(
+                    "error: cannot load config `{}`: {:?}",
+                    config_path.to_string_lossy(),
+                    e,
+                ),
+            )
+        })
+    }
+
+    pub fn open<P: AsRef<Path>, U: AsRef<str>>(path: P, rpc_url: U) -> io::Result<Self> {
+        let path = path.as_ref();
+        let config_path = path.join(CONFIG_FILENAME);
+        let rpc_url = rpc_url.as_ref();
+        let (config, entries) = match fs::read_dir(path) {
+            Ok(dir_entries) => {
+                let config = Self::load_config(&config_path)?;
+                let mut entries = dir_entries
+                    .filter_map(|de| Entry::from_pathbuf(de.unwrap().path()))
```

Comment:
We could probably add some `atime`-based eviction here when we inevitably fill a disk with cache entries

Reply:
slot-based eviction?

Reply:
> slot-based eviction?

Could do that, yeah.  Like five epochs or so?

Reply:
https://github.com/solana-labs/solana/pull/15369/commits/4a658d6c519f241b321d8820d85a9a44811fbde9